### PR TITLE
More editable reblogs fixes for when HTML editor is selected by default

### PR DESCRIPTION
**Pull request by blackjackkent**
Follow-up to #503 (and thereby issue #502)

### CODE REVIEW

File: Extensions/xkit_patches.js [hunk 1]
```diff
@@ -708,26 +715,47 @@ XKit.tools.dump_config = function(){
 				},
 
 				/**
-				 * Sets the content of the post window.
+				 * Uses sneaky HTML editor magic to handle singlequote issues with editable reblogs
 				 * @param {String} new_content
 				 */
-				set_content_html_sneak: function(new_content) {
+				set_content_html_sneak: function() {
 					var html_or_markdown = 'html';
```
Comment (bvtsang) [inline Extensions/xkit_patches.js]: This is never Markdown, is it?

File: Extensions/xkit_patches.js [hunk 2]
```diff
@@ -708,26 +715,45 @@ XKit.tools.dump_config = function(){
...
-						if (html_or_markdown === "Markdown") {
-							new_content = require('to-markdown').toMarkdown(new_content);
-						}
-						if (editor_div.length === 1) {
-							var editor = window.ace.edit(editor_div[0]);
-							editor.setValue(new_content);
-							setTimeout(function(){
-								jQuery(".ace_marker-layer").empty();
-							}, 500);
+				set_content_html_sneak: function() {
```
Comment (nightpool) [inline Extensions/xkit_patches.js]: name is un-descriptive—it does a lot of stuff that doesn't have to do with "setting the content html", and in fact never sets the content html at all.

Comment (nightpool) [inline Extensions/xkit_patches.js]: maybe should just live in editable_reblogs?

File: Extensions/xkit_patches.js [hunk 3]
```diff
@@ -708,26 +715,45 @@ XKit.tools.dump_config = function(){
...
+								var editor = window.ace.edit(editor_div[0]);
+								var content = editor.getValue();
+								//tumblr_blog must be wrapped in single quotes, not double, or the dash will nom the shit out of your post
+								content = content.replace(/"tumblr_blog"/g, "'tumblr_blog'");
+								editor.setValue(content);
+								setTimeout(function(){
+									jQuery(".ace_marker-layer").empty();
+								}, 500);
+							}
+						}, true, [new_content]);
```
Comment (nightpool) [inline Extensions/xkit_patches.js]: new_content is always "", why are we passing it?

**Comment by nightpool** [comment]
In general the code duplication internally to get_editor_sneak is a little worrying to me. we should probably refactor that such that we don't have that code duplication anymore.

**Comment by nightpool** [comment]
Also, both files don't have version bumps

File: Extensions/xkit_patches.js [hunk 4]
```diff
@@ -664,10 +664,17 @@ XKit.tools.dump_config = function(){
...
+							XKit.console.add('ERROR: unable to get content html');
 							return '';
 						}
 						return content_editor.html();
+					} else {
+						var html_or_markdown = $(".tab-label[data-js-srclabel]").text();
+						if (html_or_markdown === 'HTML') {
+							return $('.ace_editor .ace_text-layer').text();
+						}
+						XKit.console.add('ERROR: unable to get content html for markdown editor');
```
Comment (nightpool) [inline Extensions/xkit_patches.js]: ?? There doesn't seem to be any way this can succeed? Am I missing something?

Comment (nightpool) [inline Extensions/xkit_patches.js]: also should be a XKit.window error, rather then a console one.

File: Extensions/xkit_patches.js (hunk 2 above)
Comment (0xazure) [inline Extensions/xkit_patches.js]: Agreed about a more descriptive name.

File: Extensions/xkit_patches.js (hunk 4 above)
Comment (blackjackkent) [inline Extensions/xkit_patches.js]: It succeeds if the editor is HTML and fails if it is Markdown. (@bvtsang figured out a way to make this work with Markdown too but it's going to take extra work and I'm going to do it separately)

Comment (blackjackkent) [inline Extensions/xkit_patches.js]: I'm leaving it as console.error for right now; I don't think that we need to yell at the user if this breaks. Can change later once this works across all editors.

Comment (bvtsang) [inline Extensions/xkit_patches.js]: To be specific, I know it's possible to make it work with Markdown, but I
haven't gotten the API call working yet.

**Comment by nightpool** [comment]
Patches still doesn't have a version bump.

code duplication is still concerning.

**Comment by nightpool** [comment]
to be clear: its expected behavior that this/#503 break ER for people who have markdown-editors by default?

**Comment by blackjackkent** [comment]
Yes. There'll be further work separately for that and we'll mention it in the announcement.

**Comment by nightpool** [comment]
OK. I think in that case we should really super-use an XKit.window. We don't want to be eating peoples reblogs by default—we should just bail out of ER altogether and warn them if they're using markdown.

(and link to https://www.tumblr.com/settings/dashboard so they can change it)

**Comment by blackjackkent** [comment]
Hrmdehrm. Yeah, okay, good point. Will change.

File: Extensions/xkit_patches.js [hunk 5]
```diff
@@ -664,10 +664,17 @@ XKit.tools.dump_config = function(){
...
 							return '';
 						}
 						return content_editor.html();
+					} else {
+						var html_or_markdown = $(".tab-label[data-js-srclabel]").text();
+						if (html_or_markdown === 'HTML') {
+							return $('.ace_editor .ace_text-layer').text();
+						}
+						XKit.window.show('Invalid editor selection', 'ERROR: XKit cannot currently get content from the markdown editor. Click <a href="https://www.tumblr.com/settings/dashboard">here</a> to edit your dashboard settings to use the rich text or HTML editors.', 'error', "<div id=\"xkit-close-message\" class=\"xkit-button\">OK</div>");
+						return '';
```
Comment (nightpool) [inline Extensions/xkit_patches.js]: still worried that this leaves ER erasing the entire content of reblogged posts. Would be best to just bail out of ER altogether. (with a try/catch maybe?)

**Comment by nightpool** [comment]
````javascript
    if (reblog_tree.length <= 0) {
        return;
    }
````
Even if there aren't any reblog_trees, there may be `class="tumblr_blog"` statements or blog links without tumblr_blog classes, so we should still make sure to run nodes.find("a[href*=tumblr]") etc. and register the submit handler

**Comment by nightpool** [comment]
Looks great!

@homu r+

File: Extensions/xkit_patches.js [hunk 6]
```diff
@@ -1,5 +1,5 @@
 //* TITLE XKit Patches **//
-//* VERSION 5.2.0 **//
+//* VERSION 5.3.0 **//
```
Comment (nightpool) [inline Extensions/xkit_patches.js]: probably could have gotten away with a PATCH bump just fyi

**Pull request merged**